Sources/TuistDependencies/Carthage/Utils/Carthage.swift

```diff
@@ -0,0 +1,94 @@
+import TSCBasic
+import TuistCore
+import TuistGraph
+import TuistSupport
+
+// MARK: - Carthage Command Generating
+
+/// Protocol that defines an interface to interact with the Carthage.
+public protocol Carthaging {
+    /// Checkouts and builds the project's dependencies
+    /// - Parameters:
+    ///   - path: Directory whose project's dependencies will be installed.
+    ///   - platforms: The platforms to build for.
+    ///   - options: The options for Carthage installation.
+    func bootstrap(at path: AbsolutePath, platforms: Set<Platform>?, options: Set<CarthageDependencies.Options>?) throws
```

Comment:
does it makes sense to even have set of platforms that are nil? Even for `options`I'd opt in for non-optional parameters as you can freely use empty set as well (and in this case I believe there is no semantic distinction)